CHANGELOG.md

```diff
@@ -1,4 +1,5 @@
 #### Changes
+* Node, Python, Java, Go: Adding support for replacing connection configured password ([#2651](https://github.com/valkey-io/valkey-glide/pull/2651))
```

Comment:
I propose to split this PR into N+1 PRs - core changes and per-client changes. That will be easier to review, test and track the changes.

Reply:
We already at the end of it so it's a bit overkill, i divided into logical commits, so each reviewer can review the relevant part. Python and node approved. Rust finished second round, and Go is going through review now.
But as for Java, at this point it should be separated.